Bottom: 3f962095baa092ca4fa554ecf821f7801b3ed0b1
Top:    da12dc815f31142ca81c3d1ee74e89fde7bae346
Author: Ben Greear <greearb@candelatech.com>
Date:   2016-04-26 21:35:48 +0530

ath10k: support up to 64 vdevs.

Fix the math for 64 vdevs bitmap.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/drivers/net/wireless/ath/ath10k/core.c b/drivers/net/wireless/ath/ath10k/core.c
index 229a6e6..b0d9cd3 100644
--- a/drivers/net/wireless/ath/ath10k/core.c
+++ b/drivers/net/wireless/ath/ath10k/core.c
@@ -2202,7 +2202,10 @@ int ath10k_core_start(struct ath10k *ar, enum ath10k_firmware_mode mode,
 		goto err_hif_stop;
 	}
 
-	ar->free_vdev_map = (1LL << ar->max_num_vdevs) - 1;
+	if (ar->max_num_vdevs >= 64)
+		ar->free_vdev_map = 0xFFFFFFFFFFFFFFFFLL;
+	else
+		ar->free_vdev_map = (1LL << ar->max_num_vdevs) - 1;
 
 	INIT_LIST_HEAD(&ar->arvifs);
